src/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/bluetooth/BluetoothManager.kt

```diff
@@ -53,7 +58,7 @@ class BluetoothManager {
                 val scanFilter = ScanFilter.Builder()
                     .setServiceData(
                       ParcelUuid(UUID.fromString(CHIP_UUID)),
-                      byteArrayOf(0, discriminator.toByte(), (discriminator shr 8).toByte())
+                      discriminator.toByteArray(3)
```

Comment:
The discriminator is the 2nd and 3rd byte, except for the upper nibble of the 3rd byte.

Not the first 3 bytes.

Reply:
#3557

Reply:
I got this from Vidhi - without it, the discovery in android does not work at all.

PR does not depend on this though - I can revert it.

Reply:
I rolled it back.